src/SymfonyCache/UserContextSubscriber.php

```diff
@@ -130,7 +130,10 @@ protected function cleanupHashLookupRequest(Request $hashLookupRequest, Request
                 $hashLookupRequest->cookies->set($name, $value);
             }
         }
-        $hashLookupRequest->headers->set('Cookie', http_build_query($sessionIds, '', '; '));
+
+        if (count($sessionIds) > 0) {
+            $hashLookupRequest->headers->set('Cookie', http_build_query($sessionIds, '', '; '));
```

Comment:
Setting the `Cookie` header is mandatory to make the response vary on it. Setting up the cookie bag is just for convenience for further down manipulation.